compiler/cli/src/org/jetbrains/kotlin/cli/jvm/compiler/KotlinCoreEnvironment.kt

```diff
@@ -646,6 +646,11 @@ class KotlinCoreEnvironment private constructor(
             DescriptorSerializerPlugin.registerExtensionPoint(project)
             FirExtensionRegistrarAdapter.registerExtensionPoint(project)
             TypeAttributeTranslatorExtension.registerExtensionPoint(project)
+
+            val additionalExtensionDescriptors = configuration.getList(CLIConfigurationKeys.ADDITIONAL_EXTENSION_DESCRIPTORS)
+            for (descriptor in additionalExtensionDescriptors) {
+                descriptor.registerExtensionPoint(project)
+            }
```

Comment:
This approach is a bit dirty. In particular, it will require additional efforts when we decide to support environments other than CLI. Also, it abuses `CompilerConfiguration` to some extent.

The easiest alternative solution is:
1. Add `interface NativePreLinkExtension` without `prepare` method, to one of the modules accessible from this code, and register this extension point here.
2. Add `interface PreLinkExtension : NativePreLinkExtension` with `prepare` method, to Native backend.
3. In `Linker.link`, get extensions for `NativePreLinkExtension` but safe-cast them to `PreLinkExtension`.

What do you think?